services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +42,47 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  if (!event) {
+    return ERR_CODE_INVALID_ARG;
+  }
 
+  if (xQueueSend(thermalMgrQueueHandle, event, 0) != pdTRUE) {
+    return ERR_CODE_QUEUE_FULL;
+  }
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  thermal_mgr_event_t osInterrupt = {
+    THERMAL_MGR_EVENT_OS_INTERRUPT,
+  };
+  thermalMgrSendEvent(&osInterrupt);
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
+  if (!pvParameters) {
+    return;
+  }
```

Comment:
A task should never return. I'd just remove this check altogether. You can assume pvParameters is valid